packages/app-store/routing-forms/lib/selectOptions.ts

```diff
@@ -35,14 +35,28 @@ export const getFieldWithOptions = <T extends Field>(field: T) => {
   }
   return {
     ...field,
-  };
+  } as typeof field & z.BRAND<"FIELD_WITH_OPTIONS">;
 };
 
+export function areSelectOptionsInLegacyFormat(
+  field: Pick<Field, "options"> & z.BRAND<"FIELD_WITH_OPTIONS">
+) {
+  const options = field.options || [];
+  return !!options.find((option) => !option.id);
+}
+
 export function getUIOptionsForSelect(field: Field) {
-  return getFieldWithOptions(field).options?.map((option) => {
+  const fieldWithOptions = getFieldWithOptions(field);
+  const options = fieldWithOptions.options || [];
+  const areOptionsInLegacyFormat = areSelectOptionsInLegacyFormat(
```

Comment:
We are supposed to fallback to label for all options if the options are in legacy format.

This caused options to the select to be not matching the value provided.